File: app/src/main/java/com/conghuy/example/MyApplication.java
package com.conghuy.example;

import android.app.Activity;
import android.app.Application;
import android.support.v4.app.Fragment;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import dagger.android.support.HasSupportFragmentInjector;

public class MyApplication extends Application implements HasActivityInjector, HasSupportFragmentInjector {
    @Inject
    DispatchingAndroidInjector<Activity> dispatchingActivityAndroidInjector;

    @Inject
    DispatchingAndroidInjector<Fragment> dispatchingFragmentAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        DaggerMyApplicationComponent.create().inject(this);
    }

    @Override
    public DispatchingAndroidInjector<Activity> activityInjector() {
        return dispatchingActivityAndroidInjector;
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return dispatchingFragmentAndroidInjector;
    }
}

File: app/src/main/java/com/conghuy/myapplication/MyApplication.kt
package com.conghuy.myapplication

import android.app.Application
import com.conghuy.myapplication.dagger.ApplicationComponent
import com.conghuy.myapplication.dagger.ApplicationModule
import com.conghuy.myapplication.dagger.DaggerApplicationComponent
import com.conghuy.myapplication.dagger.DataManager
import javax.inject.Inject


class MyApplication : Application() {
    lateinit var _ApplicationComponent: ApplicationComponent

    @Inject
    lateinit var dataManager: DataManager

    override fun onCreate() {
        super.onCreate()
        initApplicationModule()
    }

    private fun initApplicationModule() {
        _ApplicationComponent =
            DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
        _ApplicationComponent.inject(this)

    }
}